src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml

```diff
@@ -302,116 +303,117 @@
-
-                                            <StackPanel Orientation="Horizontal" Margin="0,12,0,0" Grid.Row="0">
-                                                <ToggleButton Name="IronPythonAlertsToggle"
-                                                              Width="{StaticResource ToggleButtonWidth}"
-                                                              Height="{StaticResource ToggleButtonHeight}"
-                                                              VerticalAlignment="Center"
-                                                              IsChecked="{Binding Path=HideIronPythonAlertsIsChecked}"
-                                                              Style="{StaticResource EllipseToggleButton1}"/>
-                                                <Label Content="Hide IronPython alerts" 
-                                                       Margin="10,0,0,0"
-                                                       VerticalAlignment="Center"
-                                                       Foreground="{StaticResource PreferencesWindowFontColor}"/>
-                                            </StackPanel>
-
-                                            <StackPanel Orientation="Horizontal" Margin="0,12,0,0" Grid.Row="1">
-                                                <ToggleButton Name="ShowWhitespaceToggle"
-                                                              Width="{StaticResource ToggleButtonWidth}"
-                                                              Height="{StaticResource ToggleButtonHeight}"
-                                                              VerticalAlignment="Center"
-                                                              IsChecked="{Binding Path=ShowWhitespaceIsChecked}"
-                                                              Style="{StaticResource EllipseToggleButton1}"/>
-                                                <Label Content="{x:Static p:Resources.PreferencesViewShowWhitespaceInPythonEditor}" 
-                                                       VerticalAlignment="Center"
-                                                       Margin="10,0,0,0"
-                                                       Foreground="{StaticResource PreferencesWindowFontColor}"/>
-                                            </StackPanel>
-                                        </Grid>
-                                    </StackPanel>
-                                </Expander>
-                            </Grid>
+                            <Expander x:Name="PythonExpander"
+                                        Grid.Row="0"
+                                        Style="{StaticResource MenuExpanderStyle}" 
+                                        IsExpanded="{Binding PreferencesTabs[0].ExpanderActive, Converter={StaticResource ExpandersBindingConverter}, ConverterParameter=PythonExpander}"
```

Comment:
This design is a bit strange that if we will need to cover expander state for `General` tab in the future, we will need to shift the index here..

Reply:
@QilongTang I changed the data structure to use a Dictionary, so in the XAML we can use something more describing like "PreferencesTabs[Features]" also I added the General tab entry as an item in the Dictionary. Thanks
see commit: https://github.com/DynamoDS/Dynamo/pull/11689/commits/4540d66f8358210f8468a3f2ce76482368424622

Reply:
@RobertGlobant20 Thanks, this looks way cleaner